cpp/src/replace/replace.cu

```diff
@@ -134,3 +134,125 @@ gdf_error gdf_find_and_replace_all(gdf_column*       col,
 {
   return find_and_replace_all(col, old_values, new_values);
 }
+
+template <typename Type>
+__global__
+void replace_nulls_with_scalar_kernel(int size, Type* vax_data, uint32_t* vax_valid, const Type *vay_data_scalar) 
+{
+  int tid = threadIdx.x;
+  int blkid = blockIdx.x;
+  int blksz = blockDim.x;
+  int gridsz = gridDim.x;
+
+  int start = tid + blkid * blksz;
+  int step = blksz * gridsz;
+
+  for (int i=start; i<size; i+=step) {
+    int index = i / warpSize;
+    uint32_t position = i % warpSize;
+    uint32_t is_vax_valid = vax_valid[index];
+
+    uint32_t sel_vax = (is_vax_valid >> position) & 1;
+    vax_data[i] = sel_vax? vax_data[i] : *vay_data_scalar;
+  }
+}
+
+
+template <typename Type>
+__global__
+void replace_nulls_with_column_kernel(int size, Type* vax_data, uint32_t* vax_valid, const Type *vay_data) 
+{
+  int tid = threadIdx.x;
+  int blkid = blockIdx.x;
+  int blksz = blockDim.x;
+  int gridsz = gridDim.x;
+
+  int start = tid + blkid * blksz;
+  int step = blksz * gridsz;
+
+  for (int i=start; i<size; i+=step) {
+    int index = i / warpSize;
+    uint32_t position = i % warpSize;
+    uint32_t is_vax_valid = vax_valid[index];
+
+    uint32_t sel_vax = (is_vax_valid >> position) & 1;
+    vax_data[i] = sel_vax? vax_data[i] : vay_data[i];
```

Comment:
This should use `gdf_is_valid` to check if a value is valid: https://github.com/rapidsai/cudf/blob/branch-0.5/cpp/src/utilities/cudf_utils.h#L21